### DISCUSSION TITLE
IDEA - Move networks to seperate file

### DISCUSSION THREAD

**Discussion by defido** [pcaversaccio/hardhat-project-template-ts#19] [Polls]
Just an idea to move the networks and xdeploy configs to root, as the networks is never used, but very long 179 lines 0_0, and the xdeploy is used often and is compressed into a simple single file. As always, the decision is yours. 'Make changes to the `xdeploy.config.js/ts file`'

<img width="1028" alt="image" src="https://user-images.githubusercontent.com/95591037/163694701-8cc55b4c-4fd4-4164-b660-929c1d46fee1.png">
<img width="1247" alt="image" src="https://user-images.githubusercontent.com/95591037/163694732-619cb2a0-1d60-458a-a7ca-b9319ffd885e.png">

(I just wanted to test the voting thing out, in the end your vote is 2x mine, so your decision is final :) )

**Comment by pcaversaccio** [comment]
While I do understand your arguments and I've even tested the same structure in the past already, it was confusing and over-engineered and not provided much of value-add since there should be one configuration file per main application due to simplicity reasons. What I mean by that: there is one `rc` file for e.g. mocha or prettier and _not_ multiples that are imported into one main file at the end. The same logic applies for the `hardhat.config` file: all hardhat and plugin-specific parameters should be placed into this single file (independent of the length) to avoid any confusion and keep the simplicity. This template should make the life easier for everyone and should not introduce further complexity that was unknown before.

**Reply by defido** [reply]
As always, if people want to they could do it themselves in a fork. Got to try out the voting too. 👍
